app/components/autocomplete/channel_mention/index.ts

```diff
@@ -79,13 +81,30 @@ const withTeamId = withObservables(['teamId', 'channelId'], ({teamId, channelId,
     };
 });
 
+function getMatchPattern(isSearch: boolean, database: Database) {
+    let matchPattern;
+
+    if (isSearch) {
+        matchPattern = of$(CHANNEL_MENTION_SEARCH_REGEX);
+    } else {
+        matchPattern = observeConfigBooleanValue(database, 'DelayChannelAutocomplete').pipe(map((c) => {
+            return c ? CHANNEL_MENTION_REGEX_DELAYED : CHANNEL_MENTION_REGEX;
+        }));
+    }
+
+    return matchPattern;
+}
+
 const enhanced = withObservables(['value', 'isSearch', 'teamId', 'cursorPosition'], ({value, isSearch, teamId, cursorPosition, database}: OwnProps) => {
-    const matchTerm = getMatchTermForChannelMention(value.substring(0, cursorPosition), isSearch);
+    const matchPattern = getMatchPattern(isSearch, database);
+    const matchTerm = matchPattern.pipe(map((regexp) => getMatchTermForChannelMention(value.substring(0, cursorPosition), regexp, isSearch)));
```

Comment:
0/5
```suggestion
    const matchTerm = observeMatchPattern(isSearch,  database).pipe(
        map((regexp) => getMatchTermForChannelMention(value.substring(0, cursorPosition), regexp, isSearch))
     );
```

Reply:
Yeah, I wasn't quite happy with how that looked. If this was just chaining methods like `.pipeMap`, then I'd have a way to do it, but the combination of functions is throwing me off a bit.

I was trying the change you suggested, but then I realized I could just do it like some of the other code in the file does it where `pipe` and `map` are on the same line which I like
```js
    const matchTerm = observeMatchPattern(isSearch, database).pipe(map((regexp) => {
        return getMatchTermForChannelMention(value.substring(0, cursorPosition), regexp, isSearch);
    }));
```